nixos/modules/services/web-servers/caddy/default.nix

```
@@ -360,14 +360,15 @@ in
       serviceConfig = let
         runOptions = ''--config ${configPath} ${optionalString (cfg.adapter != null) "--adapter ${cfg.adapter}"}'';
       in {
+        # Override the `ExecStart` line from upstream's systemd unit file by our own:
         # https://www.freedesktop.org/software/systemd/man/systemd.service.html#ExecStart=
         # If the empty string is assigned to this option, the list of commands to start is reset, prior assignments of this option will have no effect.
         ExecStart = [ "" ''${cfg.package}/bin/caddy run ${runOptions} ${optionalString cfg.resume "--resume"}'' ];
         # Validating the configuration before applying it ensures we’ll get a proper error that will be reported when switching to the configuration
         ExecReload = [ "" ''${cfg.package}/bin/caddy reload ${runOptions} --force'' ];
         User = cfg.user;
         Group = cfg.group;
-        ReadWriteDirectories = cfg.dataDir;
+        ReadWritePaths = [ cfg.dataDir ];
```

```suggestion
        ReadWritePaths = cfg.dataDir;
```

Why did you make this a list?

---

Because it is a list of paths, and if I'm not mistaken, merges as such. Most usages of `ReadWritePaths` in nixpkgs use the proper type, only a few use the fact that a single element is also assignable for systemd units.

---

Makes sense. I am just usually a bit cautious when changing a type, but this should be perfectly fine here.

Thanks for responding :)